photon-api/src/integTest/scala/com/linkedin/photon/ml/optimization/DistributedOptimizationProblemIntegTest.scala

```diff
@@ -374,6 +376,86 @@ class DistributedOptimizationProblemIntegTest extends SparkTestUtils {
       }
     }
 
+  /**
+   * Test the variance computation against a reference implementation in R glm
+   */
+  @Test
+  def testComputeVariancesAgainstReference(): Unit = sparkTest("testComputeVariancesAgainstReference") {
+    // Read the "heart disease" dataset from libSVM format
+    val input: RDD[LabeledPoint] = {
+      val tt = getClass.getClassLoader.getResource("DriverIntegTest/input/heart.txt")
+      val inputFile = tt.toString
+      val rawInput = sc.textFile(inputFile, 1)
+      rawInput.map(x => {
+        val y = x.split(" ")
+        val label = y(0).toDouble / 2 + 0.5
+        val features = y.drop(1).map(z => z.split(":")(1).toDouble) :+ 1.0
+        new LabeledPoint(label, DenseVector(features))
+      }).persist()
+    }
+
+    val optimizer = mock(classOf[Optimizer[DistributedGLMLossFunction]])
+    val statesTracker = mock(classOf[OptimizationStatesTracker])
+    val regContext = mock(classOf[RegularizationContext])
+    val optConfig = mock(classOf[FixedEffectOptimizationConfiguration])
+
+    doReturn(Some(statesTracker)).when(optimizer).getStateTracker
+    doReturn(regContext).when(optConfig).regularizationContext
+    doReturn(RegularizationType.NONE).when(regContext).regularizationType
+
+    val objective = DistributedGLMLossFunction(optConfig, LogisticLossFunction, treeAggregateDepth = 1)
+
+    val optimizationProblem = new DistributedOptimizationProblem(
+      optimizer,
+      objective,
+      samplerOption = None,
+      glmConstructorMock,
+      NoRegularizationContext,
+      isComputingVariances = true)
+
+    // From a prior optimization run
+    val coefficients = DenseVector(
+      -0.022306127,
+      1.299914831,
+      0.792316427,
+      0.033470557,
+      0.004679123,
+      -0.459432925,
+      0.294831754,
+      -0.023566341,
+      0.890054910,
+      0.410533616,
+      0.216417307,
+      1.167698255,
+      0.367261286,
+      -8.303806435)
+
+    // Produced by the reference implementation in R glm
+    val expected = DenseVector(
+      0.0007320271,
+      0.3204454,
+      0.05394657,
+      0.0001520536,
+      1.787598e-05,
+      0.3898167,
+      0.04483891,
+      0.0001226556,
+      0.2006968,
+      0.05705076,
+      0.1752335,
+      0.08054471,
+      0.01292064,
+      10.37188)
+
+    val actual: Vector[Double] = optimizationProblem.computeVariances(input, coefficients).get
+
+    assertEquals(actual.length, expected.length)
+    for (i <- 0 until expected.length) {
+      val relDiff = abs(actual(i) / expected(i) - 1)
+      assertTrue(relDiff < 0.001)
+    }
```

Comment:
Take a look at `VectorUtils.areAlmostEqual` - it's not quite the same thing but might be passable for this case

Reply:
I wanted to do a relative comparison here, since we have variances across different orders of magnitude.

Reply:
Sure - wasn't sure how big the difference was between R and our predictions in practice